lib/sof/cycle/parser.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require "active_support/core_ext/hash/keys"
+require "active_support/core_ext/object/blank"
+require "active_support/core_ext/object/inclusion"
+require "active_support/core_ext/hash/reverse_merge"
+require "active_support/isolated_execution_state"
+
+module SOF
+  # This class is not intended to be referenced directly.
+  # This is an internal implementation of Cycle behavior.
+  class Cycle
+    class Parser
+      extend Forwardable
+      PARTS_REGEX = /
+        ^(?<vol>V(?<volume>\d*))? # optional volume
+        (?<set>(?<kind>L|C|W|E) # kind
+        (?<period_count>\d+) # period count
+        (?<period_key>D|W|M|Q|Y)?)? # period_key
+        (?<from>F(?<from_date>\d{4}-\d{2}-\d{2}))?$ # optional from
+      /ix
+
+      def self.dormant_capable_kinds = %w[E W]
```

Comment:
:art: Would a constant be better than a class method since this is what constants are meant for?

:bulb: We should freeze this since there is no additional behavior?

Reply:
We should freeze it, yes. 
In my opinion, constants are great for internal reference but methods are better for external reference. 
I'm looking at this on mobile so I don't actually know where this is used but regardless the scope of this PR is namespace changes so that would need to be addressed in the future if necessary

Reply:
> [C]onstants are great for internal reference but methods are better for external reference.

Hmm, that's interesting. I've not heard that before. I suppose I could see that making sense if the class method was made private. At least that would strengthen the *intent* of class method being a *internal* reference.